### ISSUE TITLE
[pre-pull-request] opencv version (PIL alternative) for image.py

### ISSUE THREAD

**Issue by DSLituiev** [keras-team/keras-preprocessing#68]
I found that I need to rewrite `load_img` in `image.py` with opencv for my corner case of 16-bit images. Also [benchmarks show](https://www.kaggle.com/vfdev5/pil-vs-opencv) that python-opencv is faster than PIL. Will there be interest in incorporating opencv/cv2-based reading function as an alternative to PIL into this package?

**Comment by easilar** [comment]
I agree with you, I am already loading my images with cv2.. Why not putting this into keras ; )

**Comment by Dref360** [comment]
* This would be a major breaking change for tons of applications.
* This would be a pain for Windows users since they need Visual C++.
* This is also a pretty big library just for loading images. I don't think we need a library as powerful as OpenCV just to load images. 


As always, Keras values UX over performance, if you need to do something custom, just create a new class that inherits ImageDataGenerator.

**NOTE**: I do use OpenCV in my research.

Conclusion : I don't think we should switch PIL for cv2.

**Comment by DSLituiev** [comment]
I see. Well, as I see now, PIL is loaded as an optional module.
My proposition is to do same (optional dependency/loading) `opencv`, with a flag, like `load_img(...,driver='pil' )`, or a global switch `preprocessing.image.DRIVER='pil'`. 

Also I'm guessing many users have opencv already if they do any heavier processing/augmentation.
